src/Compilers/CSharp/Portable/Binder/ExpressionVariableFinder.cs

```diff
@@ -456,7 +456,7 @@ protected override LocalSymbol MakeDeclarationExpressionVariable(DeclarationExpr
                             nodeBinder: _enclosingBinder,
                             typeSyntax: node.Type,
                             identifierToken: designation.Identifier,
-                            kind: LocalDeclarationKind.RegularVariable,
+                            kind: LocalDeclarationKind.DeclarationExpression,
```

Comment:
That would be a very different approach from the way local declaration kinds are defined before the C# 7 constructs were introduced, and I do not advocate such a mixed design.